### PR TITLE
docs: correct api/envoy/service/auth/v2 docs

### PR DESCRIPTION
**Pull request by LukeShu**
Description: Update some documentation comments in `api/envoy/service/auth/v2/*.proto` to more accurately describe the *current* behavior (without making any judgment on whether that behavior is "correct" or desirable).

Risk Level: Low
Testing: None
Docs Changes: Inline in API protos
Release Notes: None?

Attn: @saumoh, who I believe to be the original author.

### CODE REVIEW

**Comment by LukeShu** [comment]
v2: Add DCO

File: api/envoy/service/auth/v2/external_auth.proto [hunk 1]
```diff
@@ -61,7 +61,8 @@ message OkHttpResponse {
 
 // Intended for gRPC and Network Authorization servers `only`.
 message CheckResponse {
-  // Status `OK` allows the request. Any other status indicates the request should be denied.
+  // Status `OK` allows the request. Status `UNKNOWN` causes Envoy to abort. Any other status
+  // indicates the request should be denied.
```
Comment (LukeShu) [inline api/envoy/service/auth/v2/external_auth.proto]: See also: https://github.com/envoyproxy/envoy/issues/6210

File: api/envoy/service/auth/v2/attribute_context.proto [hunk 2]
```diff
@@ -92,18 +93,20 @@ message AttributeContext {
     // The HTTP URL scheme, such as `http` and `https`.
     string scheme = 6;
 
-    // The HTTP URL query in the format of `name1=value`&name2=value2`, as it
-    // appears in the first line of the HTTP request. No decoding is performed.
+    // This field is always empty, and exists for compatibility reasons. The HTTP URL query is
+    // included in `path` field.
     string query = 7;
```
Comment (junr03) [inline api/envoy/service/auth/v2/attribute_context.proto]: can we enforce empty here? Is it worth it?

File: api/envoy/service/auth/v2/attribute_context.proto [hunk 3]
```diff
@@ -92,18 +93,20 @@ message AttributeContext {
...
+    // not submitted as part of HTTP requests; it is unknowable.
     string fragment = 8;
 
     // The HTTP request size in bytes. If unknown, it must be -1.
     int64 size = 9;
 
-    // The network protocol used with the request, such as
-    // "http/1.1", "spdy/3", "h2", "h2c"
+    // The network protocol used with the request, as enumerated by
+    // `source/common/http/headers.h:ProtocolStrings`, such as "HTTP/1.0", "HTTP/1.1", or
```
Comment (junr03) [inline api/envoy/service/auth/v2/attribute_context.proto]: nit: change to url

File: api/envoy/service/auth/v2/external_auth.proto [hunk 4]
```diff
@@ -61,7 +61,8 @@ message OkHttpResponse {
 
 // Intended for gRPC and Network Authorization servers `only`.
 message CheckResponse {
-  // Status `OK` allows the request. Any other status indicates the request should be denied.
+  // Status `OK` allows the request. Status `UNKNOWN` causes Envoy to abort. Any other status
```
Comment (junr03) [inline api/envoy/service/auth/v2/external_auth.proto]: what does abort mean here?

File: api/envoy/service/auth/v2/attribute_context.proto [hunk 5]
```diff
@@ -92,18 +93,20 @@ message AttributeContext {
...
     string scheme = 6;
 
-    // The HTTP URL query in the format of `name1=value`&name2=value2`, as it
-    // appears in the first line of the HTTP request. No decoding is performed.
+    // This field is always empty, and exists for compatibility reasons. The HTTP URL query is
+    // included in `path` field.
     string query = 7;
 
-    // The HTTP URL fragment, excluding leading `#`. No URL decoding is performed.
+    // This field is always empty, and exists for compatibility reasons. The URL fragment is
```
Comment (junr03) [inline api/envoy/service/auth/v2/attribute_context.proto]: same as above

File: api/envoy/service/auth/v2/external_auth.proto (hunk 4 above)
Comment (LukeShu) [reply api/envoy/service/auth/v2/external_auth.proto]: "abort" means "exit with a stack trace", which I guess is probably a bug (https://github.com/envoyproxy/envoy/issues/6210).

File: api/envoy/service/auth/v2/attribute_context.proto (hunk 2 above)
Comment (LukeShu) [reply api/envoy/service/auth/v2/attribute_context.proto]: What do you mean "enforce empty"?  Note that *Envoy* is what populates this, not the AuthService talking to Envoy. I don't believe that code outside of Envoy ever needs to create this.

File: api/envoy/service/auth/v2/attribute_context.proto (hunk 3 above)
Comment (LukeShu) [reply api/envoy/service/auth/v2/attribute_context.proto]: What should that URL look like? AFAICT, there's not a documentation page on ProtocolStrings.  Should it link to the source on GitHub?

Comment (junr03) [reply api/envoy/service/auth/v2/attribute_context.proto]: yeah link to source

File: api/envoy/service/auth/v2/attribute_context.proto (hunk 2 above)
Comment (junr03) [reply api/envoy/service/auth/v2/attribute_context.proto]: got it thanks.

**Comment by junr03** [comment]
@LukeShu just replied to your comments. I will approve after update.

**Comment by LukeShu** [comment]
@junr03 I just added a commit. I think I got the link syntax correct, but please verify that.

File: api/envoy/service/auth/v2/attribute_context.proto [hunk 6]
```diff
@@ -104,9 +104,11 @@ message AttributeContext {
     // The HTTP request size in bytes. If unknown, it must be -1.
     int64 size = 9;
 
-    // The network protocol used with the request, as enumerated by
-    // `source/common/http/headers.h:ProtocolStrings`, such as "HTTP/1.0", "HTTP/1.1", or
-    // "HTTP/2".
+    // The network protocol used with the request, such as "HTTP/1.0", "HTTP/1.1", or "HTTP/2".
+    //
+    // See `headers.h:ProtocolStrings
+    // <https://github.com/envoyproxy/envoy/blob/master/source/common/http/headers.h>`_ for a
```
Comment (mattklein123) [inline api/envoy/service/auth/v2/attribute_context.proto]: Sorry for the churn here but please use a `:repo:` link. Search the docs for other examples of how to do it.

Comment (LukeShu) [reply api/envoy/service/auth/v2/attribute_context.proto]: Sure thing!  I think I got it right?

**Pull request merged**